src/c4v/scraper/crawler/crawlers/base_crawler.py

```diff
@@ -0,0 +1,131 @@
+            Parameters:
+                + post_process_data : ([str]) -> [str] = function to call over the resulting set of urls. May be called in batches
+            Return:
+                List of urls from sitemap
+        """
+
+        # Url accumulator
+        urls: List[str] = []
+
+        # sitemaps to parse
+        sitemaps = self.get_sitemaps_from_index()
+
+        for sitemap in sitemaps:
+
+            # Get a sitemap from its url
+            sitemap_content = self.get_sitemap_from_url(sitemap)
+
+            # parse urls for the current sitemap
+            urls.extend(new_urls := self.parse_urls_from_sitemap(sitemap_content))
+
+            # process new urls if post process function exists
+            if post_process_data:
+                post_process_data(new_urls)
+
+        return urls
+
+    def get_sitemaps_from_index(self) -> List[str]:
+        """
+            Some sites may have its sitemap paginated in more sitemaps.
+            I such case, override this class. By default, it assumes 
+            the sitemap set is actually the starting one.
+        """
+        assert self.start_sitemap_url != None, "Start sitemap url not configured"
+
+        resp = requests.get(self.start_sitemap_url)
+        if resp.status_code != 200:
+            resp.raise_for_status()
+
+        return self.parse_sitemaps_urls_from_index(resp.text)
+
+    def parse_sitemaps_urls_from_index(self, sitemap_index: str) -> List[str]:
+        """
+            Get sitemap list from the content of an xml file with the filemap data
+            Parameters:
+                + sitemap_index : str = sitemap index xml content as string
+            Return:
+                List of urls listed by this index
+        """
+        soup = bs4.BeautifulSoup(sitemap_index, "xml")
+        urls = map(lambda l: l.get_text(), soup.find_all("loc"))
+        urls = filter(self.check_sitemap_url, urls)
+
+        return list(urls)
+
+    def get_sitemap_from_url(self, sitemap_url: str) -> str:
+        """
+            Get sitemap xml content from its corresponding url
+            Parameters:
+                + sitemap_url : str = sitemap url to retrieve
+            Return:
+                sitemap content if everything went ok
+        """
+        resp = requests.get(sitemap_url)
+        if resp.status_code != 200:
+            resp.raise_for_status()
+
+        return resp.text
+
+    def parse_urls_from_sitemap(self, sitemap: str) -> List[str]:
+        """
+            Given a sitemap body, parse site urls from it 
+            Parameters:
+                + sitemap : str = sitemap xml content as string
+            Return:
+                list of urls parsed for this sitemap
+        """
+        soup = bs4.BeautifulSoup(sitemap, "xml")
+        urls = map(lambda l: l.get_text(), soup.select("url > loc"))
+        urls = filter(self.check_page_url, urls)
+
+        return list(urls)
+
+    @staticmethod
```

Comment:
If you want a subclass to be able to override this behavior, then you have to use classmethod instead of staticmethod. Static methods are immutable behaviors of the type, while classmethod are behaviors that can mutate per sub-type.

Reply:
But if I use a classsmethod, then I won't be able to call this function inside the class as I need to:

```
class MyClass:
    def __init__(self):
        pass

    @classmethod
    def f(cls, x):
        print("hello from classmethod, x is ", x)

    def g(self, x):
        self.f(x)

c = MyClass()
c.g()
```

This code will crash with the following error:

```
Traceback (most recent call last):
  File "test.py", line 13, in <module>
    c.g()
TypeError: g() missing 1 required positional argument: 'x'
```

How should I access classmethods inside the same class?
I could just turn it into a regular method, the thing is that then this method won't be able to run without an instance of the class, but I think that's not such a big loss